packages/fetchai/protocols/fipa/dialogues.py

```diff
@@ -244,20 +244,12 @@ def is_permitted_for_new_dialogue(self, fipa_msg: Message) -> bool:
         - have the correct msg id and message target
         - have msg counterparty set.
 
-        :param message: the fipa message
+        :param fipa_msg: the fipa message
 
         :return: a boolean indicating whether the message is permitted for a new dialogue
         """
-        fipa_msg = cast(FipaMessage, fipa_msg)
-        this_message_id = fipa_msg.message_id
-        this_target = fipa_msg.target
-        this_performative = fipa_msg.performative
-
-        result = (
-            this_message_id == FipaDialogue.STARTING_MESSAGE_ID
-            and this_target == FipaDialogue.STARTING_TARGET
-            and this_performative == FipaMessage.Performative.CFP
-        )
+        empty_dialogue = self.empty_dialogue()
+        result = empty_dialogue.is_valid_next_message(fipa_msg)
```

Comment:
Does the empty dialogue perform checks equivalent to the LHS?

Reply:
The call to `is_valid_next_message` must do that for the empty dialogue.

Reply:
Its call to `is_valid_next_message` does the checks